lib/ansible/modules/cloud/azure/azure_rm_managed_disk.py

```diff
@@ -263,13 +263,14 @@ def exec_module(self, **kwargs):
         # unmount from the old virtual machine and mount to the new virtual machine
         vm_name = parse_resource_id(disk_instance.get('managed_by', '')).get('name') if disk_instance else None
         if self.managed_by != vm_name:
-            changed = True
-            if not self.check_mode:
-                if vm_name:
-                    self.detach(vm_name, result)
-                if self.managed_by:
-                    self.attach(self.managed_by, result)
-                result = self.get_managed_disk()
+            if vm_name:
```

Comment:
what if self.managed_by = None when vm_name, eg. user run playbook repeat with vm_name set, but managed_by not set at second run